libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp

```diff
@@ -1028,6 +1031,104 @@ void NavEKF3_core::writeExtNavVelData(const Vector3f &vel, float err, uint32_t t
     storedExtNavVel.push(extNavVelNew);
 }
 
+/*
+  update the GPS selection
+ */
+void NavEKF3_core::update_gps_selection(void)
+{
+    auto &gps = AP::gps();
+
+    // in normal operation use the primary GPS
+    selected_gps = gps.primary_sensor();
```

Comment:
"active", "selected", "preferred" and "primary".. that's a lot of words to keep track of..

Reply:
indeed, we can do a separate 'cleansing' PR later